### PR TITLE
Add NMSTATE_PIN knob to Makefile

### PR DESCRIPTION
**Pull request by qinqon**
**Is this a BUG FIX or a FEATURE ?**:

> Uncomment only one, leave it on its own line:
>
> /kind bug
> /kind enhancement

**What this PR does / why we need it**:
Future versions of nmstate/nm need to be tested to ensure that they
don't introduce something really incompatible with kubernetes-nmstate to
implement that a new "knob" is added to Makefile in the form of a
NMSTATE_PIN variable that will use future versions if content is
"future".

**Special notes for your reviewer**:

**Release note**:

```release-note
NONE
```

### CODE REVIEW

**Comment by qinqon** [comment]
Future lane is working fine https://prow.apps.ovirt.org/view/gcs/kubevirt-prow/pr-logs/pull/nmstate_kubernetes-nmstate/654/pull-kubernetes-nmstate-e2e-handler-k8s-future/1334831795899731968#1:build-log.txt%3A95

And current lane is select it correctly too https://prow.apps.ovirt.org/view/gcs/kubevirt-prow/pr-logs/pull/nmstate_kubernetes-nmstate/654/pull-kubernetes-nmstate-e2e-handler-k8s/1334826384891056129#1:build-log.txt%3A726

**Comment by qinqon** [comment]
[force-push](https://github.com/nmstate/kubernetes-nmstate/compare/c9c672080624821e359cb181e1da295f1952806d..163800e8ee266061a1321977525f2b68c34bd8dd): Rebased master now "current" is 0.3/1.26 and "future" is 0.4/1.28

**Comment by qinqon** [comment]
[force-push](https://github.com/nmstate/kubernetes-nmstate/compare/163800e8ee266061a1321977525f2b68c34bd8dd..48e15e51cc648e87e4b03448aebd63595b95529d): Moved NM_FUTURE_VERSION to 1.26

File: build/Dockerfile [hunk 1]
```diff
@@ -1,8 +1,10 @@
 FROM centos:8
-
+ARG NMSTATE_COPR_REPO
+ARG NM_COPR_REPO
 RUN dnf install -b -y dnf-plugins-core && \
-    dnf copr enable -y networkmanager/NetworkManager-1.26 && \
-    dnf copr enable -y nmstate/nmstate-0.3 && \
+    dnf copr enable -y networkmanager/$NM_COPR_REPO && \
+    dnf copr enable -y nmstate/$NMSTATE_COPR_REPO && \
+    dnf copr enable -y nmstate/nispor && \
```
Comment (phoracek) [inline build/Dockerfile:7]: nispor needs to have a matching version too, don't they release versioned copr?

File: Makefile [hunk 2]
```diff
@@ -4,6 +4,20 @@ IMAGE_REGISTRY ?= quay.io
 IMAGE_REPO ?= nmstate
 NAMESPACE ?= nmstate
 
+NMSTATE_CURRENT_COPR_REPO=nmstate-0.3
+NM_CURRENT_COPR_REPO=NetworkManager-1.26
+
+NMSTATE_FUTURE_COPR_REPO=nmstate
```
Comment (oshoval) [inline Makefile:10]: maybe worth to allow overriding those argument from cli ?

File: build/Dockerfile (hunk 1 above)
Comment (qinqon) [reply build/Dockerfile:7]: Nope they don't have versioned copr yet.

File: Makefile (hunk 2 above)
Comment (qinqon) [reply Makefile:10]: We don't want that, we just want to put the release on Makefile and open the door to test know future and current.

File: Makefile [hunk 3]
```diff
@@ -4,6 +4,20 @@ IMAGE_REGISTRY ?= quay.io
...
+NM_FUTURE_COPR_REPO=NetworkManager-1.26
+
+ifeq ($(NMSTATE_PIN), future)
+	NMSTATE_COPR_REPO=$(NMSTATE_FUTURE_COPR_REPO)
```
Comment (rhrazdil) [inline Makefile:14]: We could indent other ifeq/else blocks in this Makefile as well, but that doesn't concern this PR.

**Review by rhrazdil — APPROVED** [review]
/lgtm

**Pull request merged**